sunpy/net/dataretriever/tests/test_stereo.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module tests STEREO CLient.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+import pytest
+
+from sunpy.time.timerange import TimeRange
+from sunpy.net.vso.attrs import Time, Instrument, Source, Detector
+from sunpy.net.dataretriever.client import QueryResponse
+from sunpy.net.dataretriever.downloader_factory import UnifiedResponse
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+import sunpy.net.dataretriever.sources.stereo as stereo
+
+SECCHIClient = stereo.SECCHIClient()
+PClient = stereo.PLASTICClient()
+IClient = stereo.IMPACTClient()
+SWClient = stereo.SWAVESClient()
+
+@pytest.mark.online
+@pytest.mark.parametrize("timerange, source, detector, url_start, url_end",
+                         [(TimeRange('2008/3/2 17:45:00', '2008/3/2 18:00:00'), 'STEREO_A', 'euvi',
+                          'http://stereo-ssc.nascom.nasa.gov/data/beacon/ahead/secchi/img/euvi/20080302/20080302_174530_n7euA.fts',
+                         'http://stereo-ssc.nascom.nasa.gov/data/beacon/ahead/secchi/img/euvi/20080302/20080302_175530_n7euA.fts')])
+def test_get_url_for_timerange(timerange, source, detector, url_start, url_end):
+    urls = SECCHIClient._get_url_for_timerange(timerange, source=source, detector=detector)
+    assert isinstance(urls, list)
+    assert urls[0] == url_start
+    assert urls[-1] == url_end
+
+trange = Time('2008/3/2 17:45:00', '2008/3/2 18:00:00')
+def test_can_handle_query():
+    assert SECCHIClient._can_handle_query(trange, Instrument('secchi'), Source('STEREO_A'), Detector('euvi'))
+    assert SECCHIClient._can_handle_query(trange, Instrument('secchi'), Source('STEREO_B'), Detector('hi_1'))
+    assert PClient._can_handle_query(trange, Instrument('plastic'), Source('STEREO_A'))
+    assert IClient._can_handle_query(trange, Instrument('impact'), Source('STEREO_B'))
+    assert not SWClient._can_handle_query(trange, Instrument('swaves'))
+    assert not SECCHIClient._can_handle_query(trange, Instrument('secchi'), Source('STEREO_A'), Detector('xyz'))
```

Comment:
you have been using `parametrize` in other tests, the idea of that is that you can crate a single function to run multiple tests with different inputs/outputs. As it is now, if the first assert fails we don't know whether the others are ok, and it makes it difficult to debug, because you have to run the test every single time. with a parametrize option you can creat it once and py.test will run all the tests, and you will be able to see all that fails.